android/src/org/ppsspp/ppsspp/NativeActivity.java

```diff
@@ -296,7 +296,12 @@ public void Initialize() {
 	    File sdcard = Environment.getExternalStorageDirectory();
 
 	    String externalStorageDir = sdcard.getAbsolutePath();
-	    String dataDir = this.getFilesDir().getAbsolutePath();
+	    File filesDir = this.getFilesDir();
+		String dataDir = null;
+	    if (filesDir != null) {
+	    	// Null has been seen in Google Play stacktraces
+			dataDir = this.getFilesDir().getAbsolutePath();
```

Comment:
Should we use `filesDir.getAbsolutePath()`?  Not that it matters much.

Could this maybe happen when we don't have storage access, I guess?

-[Unknown]

Reply:
Ah yeah, that's nicer.

No, the datadir is a private directory that should always be accessible. All I can think of is platform bugs combined with disk full.